spec/unit/matrix-client.spec.ts

```diff
@@ -104,7 +134,14 @@ describe("MatrixClient", function() {
         method: string;
         path: string;
     } | null = null;
-    function httpReq(method, path, qp, data, prefix) {
+    function httpReq(
+        method: Method,
+        path: string,
+        queryParams?: QueryDict,
+        body?: Body,
+        requestOpts: IRequestOpts = {},
+    ) {
```

Comment:
Updated to match the actual thing:

https://github.com/matrix-org/matrix-js-sdk/blob/d692a5dbe239d06e04159c40f282159725415689/src/http-api/fetch.ts#L137-L143

Reply:
Looks like I bunch of this was also done in https://github.com/matrix-org/matrix-js-sdk/pull/2895